source/common/ssl/context_impl.cc

```diff
@@ -152,6 +152,30 @@ ContextImpl::ContextImpl(ContextManagerImpl& parent, Stats::Scope& scope,
     }
   }
 
+  if (!config.certificateRevocationList().empty()) {
+    bssl::UniquePtr<BIO> bio(
+        BIO_new_mem_buf(const_cast<char*>(config.certificateRevocationList().data()),
+                        config.certificateRevocationList().size()));
+    RELEASE_ASSERT(bio != nullptr);
+
+    // Based on BoringSSL's X509_load_cert_crl_file().
+    bssl::UniquePtr<STACK_OF(X509_INFO)> list(
+        PEM_X509_INFO_read_bio(bio.get(), nullptr, nullptr, nullptr));
+    if (list == nullptr) {
+      throw EnvoyException(
+          fmt::format("Failed to load CRL from {}", config.certificateRevocationListPath()));
+    }
+
+    X509_STORE* cs = SSL_CTX_get_cert_store(ctx_.get());
+    for (const X509_INFO* item : list.get()) {
+      if (item->crl) {
+        X509_STORE_add_crl(cs, item->crl);
+      }
+    }
+
+    X509_STORE_set_flags(cs, X509_V_FLAG_CRL_CHECK);
```

Comment:
`X509_V_FLAG_CRL_CHECK|X509_V_FLAG_CRL_CHECK_ALL`

Reply:
Do we actually want to enable `X509_V_FLAG_CRL_CHECK_ALL`?  I had strange failures when turning that flag on during testing, and there's [at least one bug][bug] still open about it.

[bug]: https://github.com/openssl/openssl/issues/5081

Reply:
I'm pretty sure we want it. What were the error, do you recall?